singer/src/test/java/com/pinterest/singer/writer/kafka/TestCommittableKafkaWriter.java

```diff
@@ -292,4 +305,159 @@ public void testWriterWithHeadersInjectorEnabledWithWrongClass() throws Exceptio
+  private void populateLogMessagesForCorruptedMessagesTests(List<LogMessage> logMessages, List<PartitionInfo> partitions) {
+    HashFunction crc32 = Hashing.crc32();
+    CRC32 crcChecksum = new CRC32();
+    long session = System.currentTimeMillis();
+    for (int i = 0; i < NUM_EVENTS; i++) {
+      LogMessage logMessage = new LogMessage();
+      logMessage.setKey(ByteBuffer.allocate(100).put(String.valueOf(i).getBytes()));
+      logMessage.setMessage(ByteBuffer.allocate(100).put(String.valueOf(i).getBytes()));
+
+      int partitionId = Math.abs(crc32.hashBytes(logMessage.getKey()).asInt() % partitions.size());
+
+      // set headers
+      LoggingAuditHeaders headers = new LoggingAuditHeaders()
+              .setHost("host-name")
+              .setLogName("topicx")
+              .setPid(partitionId)
+              .setSession(session);
+      logMessage.setLoggingAuditHeaders(headers);
+
+      // compute and set crc checksum
+      crcChecksum.reset();
+      crcChecksum.update(logMessage.getMessage());
+      logMessage.setChecksum(crcChecksum.getValue());
+
+      logMessages.add(logMessage);
+    }
+  }
+
+  @Test
+  public void testWriterWithSkipCorruptedMessageDeletionEnabled() throws Exception {
+    CommittableKafkaWriter writer = setupWriterForCorruptedMessagesTests(true);
+
+    List<PartitionInfo> partitions = ImmutableList.copyOf(Arrays.asList(
+            new PartitionInfo("topicx", 1, new Node(2, "broker2", 9092, "us-east-1b"), null, null),
+            new PartitionInfo("topicx", 0, new Node(1, "broker1", 9092, "us-east-1a"), null, null),
+            new PartitionInfo("topicx", 2, new Node(3, "broker3", 9092, "us-east-1c"), null, null),
+            new PartitionInfo("topicx", 6, new Node(2, "broker2", 9092, "us-east-1b"), null, null),
+            new PartitionInfo("topicx", 3, new Node(4, "broker4", 9092, "us-east-1a"), null, null),
+            new PartitionInfo("topicx", 5, new Node(1, "broker1", 9092, "us-east-1a"), null, null),
+            new PartitionInfo("topicx", 7, new Node(3, "broker3", 9092, "us-east-1c"), null, null),
+            new PartitionInfo("topicx", 4, new Node(5, "broker5", 9092, "us-east-1b"), null, null),
+            new PartitionInfo("topicx", 8, new Node(4, "broker4", 9092, "us-east-1a"), null, null),
+            new PartitionInfo("topicx", 9, new Node(5, "broker5", 9092, "us-east-1b"), null, null),
+            new PartitionInfo("topicx", 10, new Node(1, "broker1", 9092, "us-east-1a"), null, null)));
+    when(producer.partitionsFor("topicx")).thenReturn(partitions);
+    when(producer.send(any())).thenReturn(ConcurrentUtils.constantFuture(
+            new RecordMetadata(new TopicPartition("topicx", 0), 0L, 0L, 0L, 0L, 0, 0)));
+
+    List<LogMessage> logMessages = new ArrayList<>();
+    populateLogMessagesForCorruptedMessagesTests(logMessages, partitions);
+
+    List<byte[]> corruptedMessageKeys = new ArrayList<byte[]>();
+    for (int i = NUM_EVENTS - 1; i > NUM_EVENTS - 10; i--) {
```

Comment:
If we corrupting a constant number of events shouldn't that constant number be used in the assert statement?